### ISSUE TITLE
Tycho build fails to compile ide project

### ISSUE THREAD

**Issue by dhuebner** [eclipse/xtext-eclipse#272]
Version: Oxygen Release Candidate 1 (4.7.0 RC1)
Build id: 20170525-1839
Xtext 2.12.0 Release 
M2E 1.8.0
Java 1.8.0

- Create new xtext project using the wizard with p2 and maven stuff
- Run maven build using m2e 

``` 
[INFO] 
[INFO] --- tycho-compiler-plugin:1.0.0:compile (default-compile) @ org.xtext.example.mydsl.ide ---
[INFO] Compiling 7 source files to /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/target/classes
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] org.xtext.example.mydsl.parent ..................... SUCCESS [  0.085 s]
[INFO] org.xtext.example.mydsl ............................ SUCCESS [ 18.965 s]
[INFO] org.xtext.example.mydsl.ide ........................ FAILURE [  1.318 s]
[INFO] org.xtext.example.mydsl.ui ......................... SKIPPED
[INFO] org.xtext.example.mydsl.target ..................... SKIPPED
[INFO] org.xtext.example.mydsl.feature .................... SKIPPED
[INFO] org.xtext.example.mydsl.repository ................. SKIPPED
[INFO] org.xtext.example.mydsl.tests ...................... SKIPPED
[INFO] org.xtext.example.mydsl.ui.tests ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 58.250 s
[INFO] Finished at: 2017-05-29T09:50:57+02:00
[INFO] Final Memory: 118M/964M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:1.0.0:compile (default-compile) on project org.xtext.example.mydsl.ide: Compilation failure: Compilation failure:
[ERROR] /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/src-gen/org/xtext/example/mydsl/ide/contentassist/antlr/MyDslParser.java:[12]
[ERROR] import org.xtext.example.mydsl.services.MyDslGrammarAccess;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The type 'MyDslGrammarAccess' is not API (restriction on classpath entry '/Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/target/classes')
[ERROR] /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/src-gen/org/xtext/example/mydsl/ide/contentassist/antlr/MyDslParser.java:[17]
[ERROR] private MyDslGrammarAccess grammarAccess;
[ERROR] ^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The type 'MyDslGrammarAccess' is not API (restriction on classpath entry '/Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/target/classes')
[ERROR] /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/src-gen/org/xtext/example/mydsl/ide/contentassist/antlr/MyDslParser.java:[34]
[ERROR] put(grammarAccess.getGreetingAccess().getGroup(), "rule__Greeting__Group__0");
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The method 'MyDslGrammarAccess.getGreetingAccess()' is not API (restriction on classpath entry '/Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/target/classes')
[ERROR] /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/src-gen/org/xtext/example/mydsl/ide/contentassist/antlr/MyDslParser.java:[34]
[ERROR] put(grammarAccess.getGreetingAccess().getGroup(), "rule__Greeting__Group__0");
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The method 'MyDslGrammarAccess.GreetingElements.getGroup()' is not API (restriction on classpath entry '/Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl/target/classes')
[ERROR] /Users/dhuebner/Downloads/Eclipse.app/ws/org.xtext.example.mydsl.parent/org.xtext.example.mydsl.ide/src-gen/org/xtext/example/mydsl/ide/contentassist/antlr/MyDslParser.java:[35]
[ERROR] put(grammarAccess.getModelAccess().getGreetingsAssignment(), "rule__Model__GreetingsAssignment");
[ERROR] ^^^^^^
```

**Comment by cdietrich** [comment]
hmmm i wonder how this can happen. the services package is exported. running the maven build a second time seems to fix the issue. maybe tycho does some wired calculation of packages and cannot handle them beeing generated on the fly

**Comment by dhuebner** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=480097

**Comment by dhuebner** [comment]
Also strange: After wizard created the projects and try running Clean-Build ws
Building workspace remains in sleeping state. So you can not run m2e build anymore because  build job block user actions. 

![bildschirmfoto 2017-05-29 um 10 27 06](https://cloud.githubusercontent.com/assets/454322/26542300/acc4850a-4459-11e7-900b-019902459967.png)

**Comment by cdietrich** [comment]
hmmm the code is still there. so it looks like they broke something with tycho 1.0.0. can you verify it works with a new project and changing tycho version back to 0.26.0

**Comment by cdietrich** [comment]
https://dev.eclipse.org/mhonarc/lists/tycho-user/msg07654.html

**Comment by cdietrich** [comment]
indeed adding `<useProjectSettings>false</useProjectSettings>` fixes the issue

**Comment by dhuebner** [comment]
With 0.26.0 it works fine. 
useProjectSettings as described in wiki doesn't make sense to me. My WS settings says ERROR for restricted API. The project settings doesn't declare any validation severities at all. So the only reason I can imagine is that tycho interprets the MANIFEST.MF from the runtime project too early, before we add the export package directives

**Comment by cdietrich** [comment]
yes that was the original issue. the problem is that the `useProjectSettings` somehow overrides the `err:-forbidden` configuration for the compiler since it how takes the config from .settings files

**Comment by cdietrich** [comment]
@dhuebner @svenefftinge which solution do you prefer:

- add `<useProjectSettings>false</useProjectSettings>` next to the current workaround in parent pom
- return to tycho `0.26.0`
- change the generated `.settings` files in the wizard

**Comment by dhuebner** [comment]
@cdietrich 

> add `<useProjectSettings>false</useProjectSettings>` 

 don't like the idea at all 

> return to tycho 0.26.0

is an option, but I think the users should use the most recent stable version to get a better feedback and support from the Tycho community and us 

> change the generated .settings files in the wizard

Well... It sounds like a good idea. I would like to know how it would behave when user turns on project specific settings for Error/Warnings in eclipse 

I also think it's not a good idea to build such kinds of workarounds. Fixing the real issue would be better. But nobody seems to know how.

**Comment by cdietrich** [comment]
well `useProjectSettings` basically restores the tycho 0.26.0 behaviour with tycho 1.0.0

**Comment by dhuebner** [comment]
It will make impossible to use the project settings than, which is a good practice. 
Without further configuration on the maven site for sure

**Comment by cdietrich** [comment]
if i understand the behaviour

(1) the user creates the project once
(2) the user runs the project once
(3) the user edits the parent pom and removes the current line and the new useProjectSettings
(4) everything works fine

**Comment by dhuebner** [comment]
>  the user edits the parent pom and removes the current line and the new useProjectSettings

Where would you put this information? In the pom.xml, eclipse Help or directly on the Xtext website?

**Comment by cdietrich** [comment]
as with the current workaround, it is already in the pom

```
<plugin>
					<!-- 
						Can be removed after first generator execution
						https://bugs.eclipse.org/bugs/show_bug.cgi?id=480097
					-->
					<groupId>org.eclipse.tycho</groupId>
					<artifactId>tycho-compiler-plugin</artifactId>
					<version>${tycho-version}</version>
					<configuration>
						<compilerArgument>-err:-forbidden</compilerArgument>
						<useProjectSettings>false</useProjectSettings>
					</configuration>
				</plugin>
```

**Comment by cdietrich** [comment]
fixed in 2.13
